### ISSUE TITLE
Show interactive module dependency graph

### ISSUE THREAD

**Issue by hikerpig** [smelukov/statoscope#37]
Often I want to analyze and visualize module dependencies in a project. And I also have the need of interactively exploring that graph, for example I want to query and select matched module, and expect to see the graph changes its focused nodes to the selected modules.

[webpack analyze](http://webpack.github.io/analyse) has something very close to what I expect, with only the absence of interactive query.

Just wondering, do you have plans for this already?

If not, I would very much like to add a page for this concept, but I'm not very familiar with discoveryjs, one shallow question I have in mind is: could I make the canvas view not rerender when location hash changes?

Hopefully I raise an issue to encourage more discussions for this feature.

**Comment by smelukov** [comment]
> Just wondering, do you have plans for this already?

Hi! What kind of query do you want to do? Just module resource?

**Comment by smelukov** [comment]
> could I make the canvas view not rerender when location hash changes?

@la

**Comment by smelukov** [comment]
> could I make the canvas view not rerender when location hash changes?

@lahmatiy Can you please answer to this part of the question?

**Comment by hikerpig** [comment]
> > Just wondering, do you have plans for this already?
> 
> Hi! What kind of query do you want to do? Just module resource?

Yes, modules and packages. So I will be able to cut unreasonable dependency chains and figure out a better structure.

**Comment by nickshevr** [comment]
@hikerpig Why do u want to analyze graph with thousand nodes with your eyes? It's really hard.

**Comment by hikerpig** [comment]
> @hikerpig Why do u want to analyze graph with thousand nodes with your eyes? It's really hard.

I expect the graph will come with an input (and maybe also some checkboxes) to filter modules interactive as I type. I guess this will become quite a big feature, though.

**Comment by ai** [comment]
I love to see this diagram for Size Limit too. It will allow to faster find the biggest file.
